client/src/manage/Substack.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { useEffect, useState } from 'react'
+import { FlexColumn, Main, Row } from '../components/Layout'
+import { BaseText, Desc, DescLeft, SmallText, Title } from '../components/Text'
+import config from '../../config'
+import { Button, Input, LinkWrarpper } from '../components/Controls'
+import { buildClient, EWSTypes } from '../api'
+import { getSld, getSubdomain } from '../utils'
+import { isValidSubstackLandingUrl } from '../../../common/substack-utils'
+import { Feedback, Loading } from '../components/Misc'
+import useDebounce from '../hooks/useDebounce'
+import styled from 'styled-components'
+import { toast } from 'react-toastify'
+import { useTryCatch } from '../hooks/useTryCatch'
+import { ethers } from 'ethers'
+import detectEthereumProvider from '@metamask/detect-provider'
+import { type ExternalProvider } from '@ethersproject/providers'
+import { EthereumProvider } from '@walletconnect/ethereum-provider'
+
+const Container = styled(Main)`
+  margin: 0 auto;
+  padding: 0 16px;
+  max-width: 800px;
+  // TODO: responsive
+`
+
+interface SuccessWithExplorerLinkParameters {
+  message: string
+  txHash: string
+}
+
+const SuccessWithExplorerLink = ({ message, txHash }: SuccessWithExplorerLinkParameters): JSX.Element => {
+  return <FlexColumn style={{ gap: 8 }}>
+    <BaseText>{message}</BaseText>
+    <LinkWrarpper target='_blank' href={config.explorer(txHash)}>
+      <BaseText>View transaction</BaseText>
+    </LinkWrarpper>
+  </FlexColumn>
+}
+
+const SmallTextGrey = styled(SmallText)`
+  color: grey;
+`
+
+const InputBox = styled(Input)`
+  border-bottom: none;
+  font-size: 16px;
+  margin: 0;
+  background: #e0e0e0;
+  &:hover{
+    border-bottom: none;
+  }
+`
+
+const LabelText = styled(BaseText)`
+  white-space: nowrap;
+`
+
+const Manage = (): JSX.Element => {
```

Comment:
should use different component name so IDE doesn't get confused which to auto import

Reply:
Done